assets/js/index.js

```diff
@@ -198,13 +206,24 @@ function updateDisplay() {
         if (i == code) {
           selected = true;
         }
-        text += `<option value="${i}" ${i == code ? 'selected': ''}>` + messageArray[j].name + '</option>';
+        var catText = categoryMap.get(messageArray[j].category);
+        var option = `<option value="${i}"${i == code ? ' selected': ''}>` + messageArray[j].name + '</option>';
```

Comment:
Is the `undefined` check here (and below in line 222) needed, given that the `category` value is marked as required? Unless you want to protect about incorrect `category` values; or is this intended for "hidden" categories? Though we also have `hidden: true`.

Reply:
If you're referring to line 212 (?), I think it's necessary to check if `catText` exists because otherwise, it'd be tacking on text to `undefined` for the first message. 
In line 222, it's necessary because if there are no messages that fit a category, it would add an `optgroup` with no options as its content.

Reply:
Ah, sorry you are right. I misunderstood these checks. Seems fine then.